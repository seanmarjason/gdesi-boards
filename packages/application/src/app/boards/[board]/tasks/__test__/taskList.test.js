import { render } from '@testing-library/react'
import TaskList from '../taskList';

jest.mock("next/navigation");

const tasks = [{
    "id": 4,
    "boardid": 1,
    "title": "Sample Task 4",
    "type": "Document",
    "assignee": 3,
    "description": "Some description",
    "links": [],
    "comments": [],
    "status": "To Do",
    "deadline": "2025-04-28T00:00:00.000Z",
    "starteddate": null,
    "completeddate": null
  },
  {
    "id": 15,
    "boardid": 1,
    "title": "Newly created task",
    "type": "Meeting",
    "assignee": 3,
    "description": "some new task",
    "links": [],
    "comments": [],
    "status": "To Do",
    "deadline": "2025-03-30T00:00:00.000Z",
    "starteddate": null,
    "completeddate": null
  },
]


global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve(tasks),
    }),
  );

it('TaskList renders correctly', async () => {
  const { container } = render(<TaskList boardId={1} />)
  expect(container).toMatchSnapshot()
})


