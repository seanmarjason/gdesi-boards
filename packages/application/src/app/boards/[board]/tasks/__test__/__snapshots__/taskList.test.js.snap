// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList renders correctly 1`] = `
<div>
  <h6
    class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
    element="h1"
  >
    Loading...
  </h6>
</div>
`;
